src/libasr/codegen/asr_to_wasm.cpp

```diff
@@ -511,29 +511,21 @@ class ASRToWASMVisitor : public ASR::BaseVisitor<ASRToWASMVisitor> {
         no_of_functions++;
     }
 
-    template <typename T>
-    bool is_unsupported_function(const T& x) {
-        std::string func_or_sub = "";
-        if (x.class_type == ASR::symbolType::Function) {
-            func_or_sub = "Function";
-        } else {
-            throw CodeGenError("has_c_function_call: C call unknown type");
-        }
+    bool is_unsupported_function(const ASR::Function_t& x) {
         if(!x.n_body) {
-            diag.codegen_warning_label(func_or_sub + " with no body", {x.base.base.loc}, std::string(x.m_name));
-            return true;
+            diag.codegen_warning_label("Function with no body", {x.base.base.loc}, std::string(x.m_name));
+            // return true;
         }
-        if (x.m_abi == ASR::abiType::BindC
-                && x.m_deftype == ASR::deftypeType::Interface) {
-                diag.codegen_warning_label("WASM: BindC and Interface " + func_or_sub + " not yet spported", { x.base.base.loc }, std::string(x.m_name));
+        if (ASRUtils::is_intrinsic_function2(&x)) {
```

Comment:
You might need to keep the previous condition "and" only intrinsic, like this:
```suggestion
        if (x.m_abi == ASR::abiType::BindC
                && x.m_deftype == ASR::deftypeType::Interface && ASRUtils::is_intrinsic_function2(&x)) {
```

Reply:
That way it will work exactly as before for all the tests, since up until now, all the `bind(c)` were intrinsic. But by adding the "intrinsic" check, we allow user defined `bind(c)` that will now work.

Reply:
Got it.

Reply:
Thank you for the guidance and support, Ondřej. It worked.